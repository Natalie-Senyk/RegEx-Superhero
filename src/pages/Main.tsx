import React, { useContext, useState, useEffect } from "react"
import Input from "../components/Input"
import classes from "./Main.module.css"
import MainImage from "../assets/main.png"
import { RegexContext } from "../store/regex-context"
import "animate.css"
import LeafImg from '../assets/wave-image.png'

const Main: React.FC = () => {
  const [btnIsHighlighted, setBtnIsHighlighted] = useState<boolean>(false)
  const regExContext = useContext(RegexContext)
  const { currentWord } = regExContext
  const { currentLevel } = regExContext
  const { numberOfGuessedWords } = regExContext


  useEffect(() => {
    if (currentLevel === 1 && numberOfGuessedWords === 0) {
      return
    }
    setBtnIsHighlighted(true)
    const timer = setTimeout(() => {
      setBtnIsHighlighted(false)
    }, 300)

    return () => {
      clearTimeout(timer)
    }
  }, [currentLevel, numberOfGuessedWords])



  return (
    <div className={classes.main}>
      <div className={classes.content}>
        <h1>
          Your current level:
          <span
            className={`${classes.badge} ${
              btnIsHighlighted ? classes.bump : ""
            }`}
          >
            {regExContext.currentLevel}
          </span>
        </h1>
        <h2>
          Number of guessed words:&nbsp;
          <span data-testid="guessed-words"
            className={`${classes.wordNumber} ${
              btnIsHighlighted ? classes.bump : ""
            } `}
          >
            {regExContext.numberOfGuessedWords}
          </span>
        </h2>
        <p className="animate__animated animate__pulse animate__delay-2s">
          Enter RegEx for <span className={classes.wordToGuess} data-testid="current-word">"{currentWord}"</span>
        </p>

        <div className={classes.formStyle} data-testid="input-block" >
          <Input/>
        </div>
      </div>
      <img className={`${classes.leaf} animate__animated animate__bounce animate__delay-1s` } src={LeafImg} alt="leaf" />
      <img className={classes.girl} data-testid="image" src={MainImage} alt="girl with cards" />
    </div>
  )
}

export default Main
