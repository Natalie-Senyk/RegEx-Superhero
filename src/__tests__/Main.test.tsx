import { render, screen } from "@testing-library/react"
import Main from "../pages/Main"
import "@testing-library/jest-dom"


describe("Main component", () => {
  beforeEach(() => {
    render(<Main />)
  })
  it("should display image on the main page", () => {
    const image = screen.getByTestId("image")
    expect(image).toBeTruthy()
  })
  it("should display main header", () => {
    const header = screen.getByText("Your current level", { exact: false })
    expect(header).toBeInTheDocument()
  })
  it("should display second header on the number of words", () => {
    const subHeader = screen.getByText("Number of guessed words", {
      exact: false,
    })
    expect(subHeader).toBeInTheDocument()
  })
  it("should display user input field", () => {
    const input = screen.getByTestId("input-block")
    expect(input).toBeInTheDocument()
  })
  it('should display the current word to guess, not just empty "" ', () => {
    const currentWord = screen.queryByTestId("current-word")
    expect(currentWord).not.toBeUndefined()
  })
//   it("should update Number of Guessed Words if the input is correct", () => {
//     const numberOfGuessedWords = screen.queryByTestId("guessed-words")
//     expect(numberOfGuessedWords).toEqual(0)
//   })
})
