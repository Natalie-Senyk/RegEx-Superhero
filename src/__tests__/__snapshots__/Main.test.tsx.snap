// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should correspond to snapshot 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="content"
    >
      <h1>
        Your current level:
        <span
          class="badge "
        >
          1
        </span>
      </h1>
      <h2>
        Number of guessed words:Â 
        <span
          class="wordNumber  "
          data-testid="guessed-words"
        >
          0
        </span>
      </h2>
      <p
        class="animate__animated animate__pulse animate__delay-2s"
      >
        Enter RegEx that matches all the words down below:
      </p>
      <div
        class="formStyle"
        data-testid="input-block"
      >
        <form
          autocomplete="off"
          class="makeStyles-root-25"
          data-testid="form"
          novalidate=""
        >
          <div
            class="MuiFormControl-root MuiTextField-root text-field makeStyles-inputField-26"
            data-testid="input-field"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink="false"
              for="filled-basic"
              id="filled-basic-label"
            >
              Enter your RegEx here
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                id="filled-basic"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="makeStyles-root-25"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-27 MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                skip
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 5H11l5 7-5 7h4.5l5-7z"
                    />
                    <path
                      d="M8.5 5H4l5 7-5 7h4.5l5-7z"
                    />
                  </svg>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-27 MuiButton-containedPrimary"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                submit
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <span
                    aria-hidden="true"
                    class="material-icons MuiIcon-root"
                  >
                    send
                  </span>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
    <img
      alt="leaf"
      class="leaf animate__animated animate__bounce animate__delay-1s"
      src="wave-image.png"
    />
    <img
      alt="girl with cards"
      class="girl"
      data-testid="image"
      src="main.png"
    />
  </div>
</DocumentFragment>
`;
